app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,31 @@
+class LikesController < ApplicationController
+  def create
+    @bookmark = Bookmark.find(params[:bookmark_id])
+    @topic = Topic.find(params[:topic_id])
+    like = current_user.likes.build(bookmark: @bookmark)
+
+    if like.save
+      flash[:notice] = "You have liked the bookmark"
+      redirect_to @topic
+    else
+      flash[:error] = "There was an error, please try again"
+    end
+  end
+
+  def destroy
+    # Get the bookmark from the params
+    @bookmark = Bookmark.find(params[:bookmark_id])
+    @topic = Topic.find(params[:topic_id])
+    # Find the current user's like with the ID in the params
+    @like = current_user.likes.where(bookmark_id: @bookmark.id).first
```

Comment:
No need for `*_id` and `.id` usage when building queries like this on line 20. Can be simplified to:

```ruby
@like = current_user.likes.where(bookmark: @bookmark).first
```

This then can be simplified further to use `find_by` instead of `where` as we only expect one object to be returned by the query:

```ruby
@like = current_user.likes.find_by(bookmark: @bookmark)
```

Good work on user-scoping these `like` queries.